lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -816,28 +1170,56 @@ void ScienceSensorsSystem::PostUpdate(const ignition::gazebo::UpdateInfo &_info,
         }
         continue;
       }
-      pcl::PointXYZ searchPoint(
-          sensorLatLon.value().X(),
-          sensorLatLon.value().Y(),
-          sensorLatLon.value().Z());
 
-      // kNN search (alternatives are voxel search and radius search. kNN
-      // search is good for variable resolution when the distance to the next
-      // neighbor is unknown).
-      int k = 8;
+      // Don't need to interpolate EVERY PostUpdate(). That's overkill.
+      // Only need to do it after robot has moved a distance from when we did
+      // the previous interpolation
+      if (sensorLatLon.value().Distance(
+          this->dataPtr->lastSensorUpdateCartesian) <
+        this->dataPtr->INTERPOLATE_DIST_THRESH)
```

Comment:
We shouldn't use this magic number. It should be related to the data density. The optimal would be to compute the bounding box of when a nearest neighbor will transition. However as a proxy some fraction of the distance to the nearest neighbor might make sense as a heuristic causing over computation but it will be data driven instead of a magic number.

Reply:
This is just the quickest way to cut down the number of times interpolation is run, so that it doesn't run every PostUpdate(). This code is not tested until we have interpolation, so this is more a preventative measure.

We can add optimization later, if speed is a problem. That requires some thought, because the data has variable density. That will be a PR in itself.

Reply:
I should add that the only reason this was introduced right now, is that none of the visualization was showing when interpolation was always called.